src/Test.UnitTests.Sarif/Baseline/ResultMatching/SarifLogResultMatcherTests.cs

```diff
@@ -359,42 +359,41 @@ public void SarifLogResultMatcher_MultipleLogsDuplicateData_WorksAsExpected()
         [Fact]
         public void SarifLogResultMatcher_PreservesPropertiesProperly()
         {
+            // Verify that SarifLog matching keeps old or latest property values on matched Results
+            // depending on how it was configured.
+
+            // This test used to verify that old Artifact properties are also preserved, but this behavior
```

Comment:
>This [](start = 15, length = 4)

I'd have to think about this more to understand why we put this behavior in originally, i.e., whether it was driven by a concrete scenario or just some fuzzy thinking. On surface, if you are allowing the oldest thing to have precedence, you might prefer to retain a sense of the artifact at the time of the oldest run. looking over the properties on the artifact object, though, I tend to agree this really doesn't make much sense. perhaps there'd be a use for knowing the last modified time at some earlier date. Possibly the 'roles' in play at the earliest date might be relevant. In any case, these remarks are entirely consistent with your observation that merging will be a careful, property-specific operation, rather than a blanket retention of one version over the other

Reply:
Yeah - it's wrong the way it was. By removing the logic and releasing, we'll find out if anyone really wanted the behavior. Since it'll be complex to fix properly (merging the latest Artifact with the corresponding older properties) it seemed better to release and find out if this matters.